game/PsyDoom/Config/ConfigSerialization_Multiplayer.cpp

```diff
@@ -0,0 +1,83 @@
+#include "ConfigSerialization_Multiplayer.h"
+
+#include "Config.h"
+#include "IniUtils.h"
+
+#include <string>
+
+BEGIN_NAMESPACE(ConfigSerialization)
+
+using namespace Config;
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+// Config field storage
+//------------------------------------------------------------------------------------------------------------------------------------------
+Config_Multiplayer gConfig_Multiplayer = {};
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+// Initialize the config serializers for multiplayer related config
+//------------------------------------------------------------------------------------------------------------------------------------------
+void initCfgSerialization_Multiplayer() noexcept {
+    auto& cfg = gConfig_Multiplayer;
+
+    cfg.noFriendlyFire = makeConfigField(
+        "NoFriendlyFire",
+        "Cooperative: Players will not take damage from other players.\n"
+        "\n"
+        "Note: players can still take damage from nearby exploding barrels\n"
+        "and being telefragged by the other player.",
+        gbNoFriendlyFire,
+        false
+    );
+
+    cfg.mpThings = makeConfigField(
+        "MPThings",
+        "Cooperative: This will add things to the map that are tagged for multiplayer only\,\n"
+        "such as: additional monsters\, weapons\, and powerups.\n"
+        "\n"
+        "Note: the additional weapons and powerups are normally only found in deathmatch.\n"
+        "The additional monsters were possibly intended for another game mode that was\n"
+        "never implemented or carried over from PC Doom and never removed.",
+        gbMPThings,
+        0
+    );
```

Comment:
@ts-miller and I were thinking about moving this setting to the Launcher tab under game options and it would apply to single player and co-op. We could put in the tool tip a note about it being similar in function to UV+, but it would apply to all difficulties.

Reply:
Sorry I missed this, I think it is a good idea. But I would keep the "Spawn Multiplayer Only Things" in the Multiplayer Tab if only for the reason this might confuse matters when it comes to people verifying playthroughs on various difficulties as there is nothing in game stating this would be on. You would be creating 4 more unofficial tiered difficulties. So I think only having an official difficulty like enable Ultra Violence+ ""Nightmare+Multiplayer Things"" on the main Launcher tab is a better option with it also stated in the main game title screen when enabled.

I do like the idea of being able to enable multiplayer things on other difficulties though. I just think it shouldn't be a main show option due what I previously mentioned.

Reply:
![psydoomui](https://user-images.githubusercontent.com/50659277/199290682-bfff221b-d9dd-4273-81f0-1d584f3cb544.JPG)
<img width="905" alt="doom menu" src="https://user-images.githubusercontent.com/50659277/199290722-2cc56f8b-82c9-4ff2-88f0-752b5672aeb3.PNG">

Bad mock-up but I think it would resemble something like this. I don't think there is a + symbol available so it would have to be in text.

Reply:
As a side not I think No Monsters and Turbo mode should be under a different tab. Maybe game?

Reply:
Items in the 'Launcher' tab all correspond to PsyDoom command line arguments, as opposed to settings in .ini files for the all other tabs. I keep them separate because I want that distinction to remain clear for those who make their own startup/launch scripts.

As far as UV+ goes this has already been discussed by #58. We can get an _effective_ UV+ however by just moving the DM items setting to be available for single player too. Doing so would also offer more flexibility, since it could be used with lower difficulty settings too (if desired).